### ISSUE TITLE
Output values mismatch when test_to_channel_last_io helper function is used to transpose the input

### ISSUE THREAD

**Issue by codewarrior26** [google-ai-edge/ai-edge-torch#147]
### Description of the bug:

It looks like the torch model's output does not match the ouput of the same model that was converted to tflite with _to_channel_last_io_  function provided [here](https://github.com/google-ai-edge/ai-edge-torch/blob/main/docs/pytorch_converter/README.md#convert-model-with-nhwc-channel-last-inputsoutputs)

To reproduce it:

`resnet18 = torchvision.models.resnet18(torchvision.models.ResNet18_Weights.IMAGENET1K_V1).eval()`
`sample_inputs = (torch.randn(1, 3, 224, 224),)`
`torch_output = resnet18(*sample_inputs)`

`edge_model = ai_edge_torch.convert(resnet18, sample_inputs) # without to_channel_last_function`

`nhwc_resnet18 = ai_edge_torch.to_channel_last_io(resnet18, args=[0]).eval() #with to_channel_last_function`
`sample_tflite_input = (torch.randn(1, 224, 224, 3),)`
`edge_model_channel_last = ai_edge_torch.convert(nhwc_resnet18, sample_tflite_input)`

`edge_output = edge_model(*sample_inputs)`
`edge_output_channel_last = edge_model_channel_last(*sample_tflite_input)`

`if np.allclose(torch_output.detach().numpy(), edge_output, atol=1e-5):`
 `   print("Inference result with Pytorch and TfLite was within tolerance")`
`else:`
`    print("Something wrong with Pytorch --> TfLite")`

This gives: **Inference result with Pytorch and TfLite was within tolerance**

But, 
`if np.allclose(edge_output, edge_output_channel_last, atol=1e-5):`
 `   print("Inference result with Pytorch and TfLite was within tolerance")`
`else:`
 `   print("Something wrong with Pytorch --> TfLite")`

gives **Something wrong with Pytorch --> TfLite**

### Actual vs expected behavior:

Actual:

The edge output with channel_last_io function does NOT match edge output without channel_last_io function.
The edge output with channel_last_io function does NOT match pytorch model output without

Expected:
Both should match.

### Any other information you'd like to share?

I visualized the models on netron. For the converted model without the helper function used, there is a transpose operator added after the input (which is expected because of channel-first to channel-last expectations). With the helper function, we see the input shape is changed to channel-last (so there is no transpose operator added - this is also expected). So in a nutshell, the conversion process looks good with the helper function used, but I'm not sure why the outputs are mismatched. Am I missing something? Please let me know. Thanks!

**Comment by pkgoogle** [comment]
Hi @codewarrior26, you are using different tensor values for your inputs, so that's why you are producing different output. You need to transpose the same tensor values so that it is a fair comparison, not create a new tensor of a different shape... please see your modified example:

```py
import ai_edge_torch
import numpy as np
import torch
import torchvision


resnet18 = torchvision.models.resnet18(torchvision.models.ResNet18_Weights.IMAGENET1K_V1).eval()
sample_inputs = (torch.randn(1, 3, 224, 224),)
torch_output = resnet18(*sample_inputs)

edge_model = ai_edge_torch.convert(resnet18, sample_inputs) # without to_channel_last_function
nhwc_resnet18 = ai_edge_torch.to_channel_last_io(resnet18, args=[0]).eval() #with to_channel_last_function

sample_tflite_input = (torch.randn(1, 224, 224, 3),)
edge_model_channel_last = ai_edge_torch.convert(nhwc_resnet18, sample_tflite_input)

edge_output = edge_model(*sample_inputs)
# edge_output_channel_last = edge_model_channel_last(*sample_tflite_input)

if np.allclose(torch_output.detach().numpy(), edge_output, atol=1e-5):
   print("Inference result with Pytorch and TfLite was within tolerance")
else:
    print("Something wrong with Pytorch --> TfLite")

sample_tensor = sample_inputs[0]
sample_tensor = torch.transpose(sample_tensor, 1, 3)
sample_tensor = torch.transpose(sample_tensor, 1, 2)

edge_output_channel_last = edge_model_channel_last(sample_tensor)

if np.allclose(edge_output, edge_output_channel_last, atol=1e-5):
   print("Inference result with Pytorch and TfLite was within tolerance")
else:
    print("Something wrong with Pytorch --> TfLite")
```

Let me know if this resolves your issue. Thanks.

**Comment by codewarrior26** [comment]
Thanks @pkgoogle, permuting the same input tensor slipped my mind, thanks for pointing that out!
